Added CustomJSONFormatter to generate Cucumber Reports for executed test cases in an event of Thread Failure

Subrahmanyam83 opened the pull request.

Commit:
Added CustomJSONFormatter to generate Cucumber Reports for executed test cases in an event of Thread Failure.

Commit:
Reformatted the code

sponte commented on .gitignore:
```diff
@@ -8,3 +8,4 @@ release.properties
 dependency-reduced-pom.xml
 buildNumber.properties
 .mvn/timing.properties
+*.iml
```
It's ok for .iml to be in the repo

sponte commented on src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java:
```diff
@@ -0,0 +1,250 @@
+package eu.evops.maven.pluins.cucumber.parallel;
```
Is there a reason why this class does not inherit from JSONFormatter class and only override what it needs?

sponte commented on src/main/java/eu/evops/maven/pluins/cucumber/parallel/CustomJSONFormatter.java:
```diff
@@ -0,0 +1,250 @@
+/**
+ * Created by n471306 on 06/07/2017.
+ */
+public class CustomJSONFormatter implements Reporter, Formatter {
```
CustomerJSONFormatter name is not descriptive, can you rename to something more meaningful for instance StreamingJSONFormatter and adding appropriate comments to indicate why it's been added?

sponte commented on src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java:
```diff
@@ -219,7 +219,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         for (String plugin : plugins) {
             String pluginName = plugin.split(":")[0];
-            if(pluginName.matches("^(json|junit)")) {
+            if(pluginName.matches("^(CustomJSONFormatter|junit)")) {
```
This code in unnecessary. You should update Merger.get method to return correct merge plugin

sponte commented on src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java:
```diff
@@ -294,11 +297,18 @@ private void generateReportForJsonFiles(File reportOutputDirectory,
-                        getReportFileName(plugin.split(":")[0]));
-                args.add(format("%s%s", plugin, threadedReportFile.getAbsolutePath()));
-            } else {
+            if(plugin.endsWith(":")&&!plugin.startsWith("json")) {
```
Again, this should not be necessary, update getReportFileName method that expects your new json reporter plugin name (absolute)

Commit:
incorporated review comments

Commit:
incorporated review comments

Commit:
incorporated review comments

Subrahmanyam83 commented:
Please provide review comments

Commit:
incorporated review comments

Commit:
StreamingJsonFormatter makes use of CucumberJsonFormatter so that we don't do copy/paste

Commit:
1. Added unit test cases to veify the Streaming JSONFrmatter
2. Added a MOJO Test for StreamingJSONFormatter.
3. Created a switch between default and Streaming JSON Formatter.

Commit:
1. Added unit test case to verify the merger for JSONs created by Streaming JSONFrmatter.
2. Added a MOJO Test for StreamingJSONFormatter.
3. Created a switch between default and Streaming JSON Formatter.

Commit:
1. Reverted back the Language level to 1.7
2. Removed SOPs.

sponte commented on src/main/java/eu/evops/maven/pluins/cucumber/parallel/Run.java:
```diff
@@ -361,6 +373,13 @@ private void setThreadCount() {
+    /* Returns the streamingFormatter value.
+    * if true, it uses the StreamingJSON Formatter
+    * else it uses the default Cucumber JSON Formatter*/
+    private boolean getStreamingFormatterValue() throws IOException {
```
Why does this throw IOException?

Commit:
Updating project to accept either json or streaming json formatters and adding a flag to allow for selection

Commit:
Merge https://github.com/Subrahmanyam83/cucumber-runner-maven-plugin into Subrahmanyam83-master

Commit:
Adjusting unit tests after changing the formatter logic (mutually exclusive formatters logging to the same location)

Commit:
Fixing unit test

Commit:
Fixing unit tests

Commit:
changes in the MOJO class to accept the correct arguements

Commit:
Added test case to forcefully kill the process running the current thread to validate the reports generated.

sponte commented on pom.xml:
```diff
@@ -1,213 +1,392 @@
+				</version>
+				<configuration>
+					<source>
+						1.8
```
Hey, this should revert back to 1.7 please

Commit:
changed the target and source java version back to 1.7

Subrahmanyam83 committed:
Update cucumber-runner-maven-plugin.iml

Commit:
code format

Commit:
Merge branch 'master' of github.com:Subrahmanyam83/cucumber-runner-maven-plugin

Commit:
Merge github.com:eu-evops/cucumber-runner-maven-plugin

sponte committed:
Adding travis badge

Commit:
Removed lamdas and made changes to the code wrt java 1.7 in STreamingJSONFormatterTest

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
added a build step in @Before

sponte committed:
Removing @before

Travis.ci already runs mvn install -DskipTests before invoking test target

Commit:
Don't need to run mvn clean for each test

sponte committed:
Adding development notes to README file
